jsapp/js/account/organization/MemberRemoveModal.tsx

```diff
@@ -0,0 +1,130 @@
+// Libraries
+import clonedeep from 'lodash.clonedeep';
+
+// Partial components
+import Button from 'jsapp/js/components/common/button';
+import InlineMessage from 'jsapp/js/components/common/inlineMessage';
+import KoboModal from 'jsapp/js/components/modals/koboModal';
+import KoboModalHeader from 'jsapp/js/components/modals/koboModalHeader';
+import KoboModalContent from 'jsapp/js/components/modals/koboModalContent';
+import KoboModalFooter from 'jsapp/js/components/modals/koboModalFooter';
+
+// Stores, hooks and utilities
+import {getSimpleMMOLabel} from './organization.utils';
+import envStore from 'jsapp/js/envStore';
+import subscriptionStore from 'jsapp/js/account/subscriptionStore';
+import {useRemoveOrganizationMember} from './membersQuery';
+import {notify} from 'alertifyjs';
+
+interface TextToDisplay {
+  title: string;
+  description: string;
+  dangerMessage: string;
+  confirmButtonLabel: string;
+}
+
+const REMOVE_SELF_TEXT: TextToDisplay = {
+  title: t('Leave this ##TEAM_OR_ORGANIZATION##'),
+  description: t('Are you sure you want to leave this ##TEAM_OR_ORGANIZATION##?'),
+  dangerMessage: t('You will immediately lose access to any projects owned by this ##TEAM_OR_ORGANIZATION##. This action cannot be undone.'),
+  confirmButtonLabel: t('Leave ##TEAM_OR_ORGANIZATION##'),
+};
+
+const REMOVE_MEMBER_TEXT: TextToDisplay = {
+  title: t('Remove ##username## from this ##TEAM_OR_ORGANIZATION##'),
+  description: t('Are you sure you want to remove ##username## from this ##TEAM_OR_ORGANIZATION##?'),
+  dangerMessage: t('Removing them from this ##TEAM_OR_ORGANIZATION## also means they will immediately lose access to any projects owned by your ##TEAM_OR_ORGANIZATION##. This action cannot be undone.'),
+  confirmButtonLabel: t('Remove member'),
+};
+
+/**
+ * @returns one of `TextToDisplay` objects with all placeholders replaced with
+ * proper values.
+ */
+function getTextToDisplay(
+  isRemovingSelf: boolean,
+  username: string,
+  mmoLabel: string
+) {
+  const text = clonedeep(isRemovingSelf ? REMOVE_SELF_TEXT : REMOVE_MEMBER_TEXT);
```

Comment:
If we move the REMOVE... texts and the replace function into the component itself could we skip this cloning? No big deal if not, just thought I'd ask.